util/arbmath/moving_average.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2023, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package arbmath
+
+// A simple moving average of a generic number type.
+type MovingAverage[T Number] struct {
+	period         int
```

Comment:
You can drop this, it's same as cap(a.buffer)

Reply:
Makes sense, I wondered about this because in some other language you're only guaranteed to get a slice of _at least_ the capacity you request, but it looks like in Go you are guaranteed to get a slice of exactly the capacity you request